src/s2e/Plugins/ExecutionMonitors/LibraryCallMonitor.h

```diff
@@ -1,82 +1,88 @@
 ///
-/// Copyright (C) 2011-2013, Dependable Systems Laboratory, EPFL
+/// Copyright (C) 2011-2017, Dependable Systems Laboratory, EPFL
+/// Copyright (C) 2016, Cyberhaven
 /// All rights reserved.
 ///
 /// Licensed under the Cyberhaven Research License Agreement.
 ///
 
-#ifndef S2E_PLUGINS_LIBCALLMON_H
-#define S2E_PLUGINS_LIBCALLMON_H
+#ifndef S2E_PLUGINS_LibraryCallMonitor_H
+#define S2E_PLUGINS_LibraryCallMonitor_H
 
 #include <s2e/CorePlugin.h>
 #include <s2e/Plugin.h>
-#include <s2e/S2EExecutionState.h>
+#include <s2e/Plugins/OSMonitors/ModuleDescriptor.h>
 
-#include <string>
-#include <tr1/unordered_map>
-#include <tr1/unordered_set>
+namespace s2e {
 
-#include <s2e/Plugins/Core/Vmi.h>
-#include <s2e/Plugins/OSMonitors/Support/ModuleExecutionDetector.h>
-#include "FunctionMonitor.h"
+class S2E;
+class S2EExecutionState;
 
-namespace s2e {
 namespace plugins {
 
+class ModuleMap;
 class OSMonitor;
+class ProcessExecutionDetector;
+class Vmi;
 
+///
+/// \brief Monitors external library function calls.
+///
+/// This plugin can be used to determine what external library calls a particular module makes at run time. The
+/// modules to monitor are those defined in the \code ProcessExecutionDetector plugin's config.
```

Comment:
I just did some tests by adding things like "kernel32.dll" and "crypt32.dll" to the moduleNames entry in the ProcessExecutionDetector config.

The answer is yes, this does allow you to track libraries imported by other libraries if they are specified in the ProcessExecutionDetector config.